File: Data Structures & Algorithms/two-integer-sum/submission-9.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hp={}
        for i in range(len(nums)):
            if nums[i] not in hp:
                hp[target-nums[i]]=i
            else:
                return [hp[nums[i]],i]
        return [-1,-1]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-9.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int len=nums.length;
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int n:nums){
            map.put(n,map.getOrDefault(n,0)+1);
        }
        int longest=0;
        for(int num:nums){
            int n=num;
            int count=0;
            if(!map.containsKey(n-1)){
                while(map.containsKey(n)){
                    count++;
                    n++;
                }
            }
            longest=Math.max(longest,count);
        }
        return longest;
    }
}
